### PR TITLE
Add checkpointing for DeepLake vector store

### PR DESCRIPTION
**Pull request by farizrahman4u**
# Add checkpointing for DeepLake vector store 

Add checkpointing to Deeplake vector store so that in case ingestion fails, data uploaded until then is available to the user.

### CODE REVIEW

**Comment by farizrahman4u** [comment]
@hwchase17

File: langchain/vectorstores/deeplake.py [hunk 1]
```diff
@@ -245,10 +247,18 @@ def ingest(sample_in: list, sample_out: list) -> None:
             elements[i : i + batch_size] for i in range(0, len(elements), batch_size)
         ]
 
+        num_workers = min(self.num_workers, len(batched) // max(self.num_workers, 1))
+
+        checkpoint_interval = self.checkpoint_interval
+        if checkpoint_interval is None:
```
Comment (dev2049) [inline langchain/vectorstores/deeplake.py:253]: could there be cases where users explicitly don't want to checkpoint (and so the None should be respected)?

**Comment by baskaryan** [comment]
cc @davidbuniat any thoughts?

**Comment by efriis** [comment]
Closing because the PR wouldn't line up with the current directory structure of the library (would need to be in /libs/langchain/langchain instead of /langchain). Feel free to reopen against the current head if it's still relevant!

**Pull request closed without merging**